index.js

```diff
@@ -317,21 +320,53 @@ const partition = (array, predicate) =>
     },
     [[], []],
   )
+const removeAndGatherComments = (keys, predicate) =>
+  keys.reduceRight(
+    (result, key) => {
+      const prev = result[0][0]
+      ;(predicate(key)
+        ? result[0].length > 0
+          ? result[1][prev] || (result[1][prev] = [])
+          : result[2]
+        : result[0]
+      ).unshift(key)
+      return result
+    },
+    // 0: non-comment keys and trailing comments
+    // 1: comments that precede a key
+    // 2: trailing comments
+    [[], {}, []],
+  )
```

Comment:
This seems too complicated to me.

```js
['//z', '//x', 'z', ' //name', 'name', 'a', '// end']
 -> keys without `//`: ['z', 'x', 'name', 'a', 'end']
 -> sort: ['a', 'end', 'name', 'x', 'z']
 -> restore: ['a', `// end`, '//name', name, '//x',''x', '//z', 'z' ]
```

WDYT?

Reply:
if anyone can think of simpler ways to do this, i'm all ears.